wpilibc/src/main/native/include/frc/kinematics/SwerveModuleState.h

```diff
@@ -25,5 +25,24 @@ struct SwerveModuleState {
    * Angle of the module.
    */
   Rotation2d angle;
+
+  /**
+   * Optimizes the angle of the module to make sure it doesn't
+   * rotate more than 90 degrees from the current state
+   */
+  SwerveModuleState OptimizeModuleAngle(
+      const SwerveModuleState& desiredState) const {
+    frc::SwerveModuleState finalStateAfterOptimization = desiredState;
+    frc::Rotation2d deltaAngle{finalStateAfterOptimization.angle.Degrees() -
+                               angle.Degrees()};
+    if (units::math::abs(deltaAngle.Degrees()) > 90_deg &&
+        units::math::abs(deltaAngle.Degrees()) < 270_deg) {
```

Comment:
Isn't a Rotation2d always between -pi and pi?

Reply:
I'm pretty sure it is but that still makes it so you rotate 180 degrees when just moving forward and back which is sub optimal and annoying to drive.

Reply:
Re-reading the Rotation2d code, turns out it doesn't normalise when given an angle value. But normalising the angle would probably make sense here.

Right now using Rotation2d here is not actually doing anything...

Reply:
The reason I was using Rotation2d was that it is part of the WPI library and is used to represent a rotation in space. I might be confused about what you're asking? By normalization you mean from -180 to 180?

Reply:
Normalization isn't the right term here. Modulus would be more accurate. Rotation2d has `value`, `cos` and `sin` members. `cos` and `sin` are used to handle modulus addition and subtraction operations without needing fmod. For example,
```cpp
Rotation2d& Rotation2d::operator+=(const Rotation2d& other) {
  double cos = Cos() * other.Cos() - Sin() * other.Sin();
  double sin = Cos() * other.Sin() + Sin() * other.Cos();
  m_cos = cos;
  m_sin = sin;
  m_value = units::radian_t(std::atan2(m_sin, m_cos));
  return *this;
}
```
If the constructor is given an angle above 2pi radians or below -2pi radians, it won't take the modulus of it, and that's by design. If you want that behavior, use the `cos` and `sin` components instead.

Reply:
Hang on, `finalStateAfterOptimization.angle` is a Rotation2d, right? Wouldn't it make sense to just directly subtract here?